Cannot query field "admin_notification" on type "Subscription"  when multiple projects defined in .graphqlrc.yml

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
Here is my `.graphqlrc.yml`:
```
projects:
  first:
    schema: ./first.graphql
    documents: ./src/first-gql.js
  second:
    schema: ./second.graphql
    documents: ./src/second-gql.js
```

Here is my `./src/first-gql.js` file:
```
const queryUsers = gql`
  query users($first: Int) {
    users(first: $first) {
      id
      symbol
    }
  }
`;
```

Here is my `./src/second-gql.js` file:
```
const subscribeAdminNotification = gql`
  subscription admin_notification(
    $limit: Int
    $order_by: [admin_notification_order_by!]
    $where: admin_notification_bool_exp
  ) {
    admin_notification(limit: $limit, order_by: $order_by, where: $where) {
      id
      level
    }
  }
`;
```
When running `npx eslint --ext graphql,js .` the `graphql-eslint` linter complains that `Cannot query field "admin_notification" on type "Subscription"`, because it is looking in the wrong `.graphql` schema!!!

If I remove the first schema definition from `.graphqlrc.yml`, then it works OK.


**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
`graphql-eslint` should lint .js files against their related .graphql schema definition (as defined by the `documents` field in `.graphqlrc.yml`)

Otherwise, it breaks compatibility with `graphql-config@3` which documentation states that [it's supporting multiple projects](https://graphql-config.com/usage#projects)

**May I know how do you lint when you have multiple .graphql files/projects in `.graphqlrc.yml`?**

Thanks for your help!

Well, actually directly linting the documents file works: `npx eslint ./query2.js`

BUT linting all files with `npx eslint --ext graphql,js .` still throw the error:

```
/home/me/code/github.com/waterdrop01/graphql-eslint-prettier/query2.js
  5:6  error  Cannot query field "book" on type "Query"  @graphql-eslint/fields-on-correct-type

✖ 1 problem (1 error, 0 warnings)
```

I created a [repository](https://github.com/waterdrop01/graphql-eslint-prettier) so you can reproduce the issue.

--- a/.changeset/real-pigs-develop.md
+++ b/.changeset/real-pigs-develop.md
@@ -1,0 +1,5 @@
+---
+'@graphql-eslint/eslint-plugin': patch
+---
+
+fix(graphql-config): pass real filepath to `gqlConfig.getProjectForFile`

--- a/packages/plugin/src/schema.ts
+++ b/packages/plugin/src/schema.ts
@@ -35,14 +35,14 @@ export function getSchema(options: ParserOptions, gqlConfig: GraphQLConfig): Gra
   // We are using the directory of the file as the key for the schema caching, to avoid reloading of the schema.
   if (gqlConfig && options.filePath) {
     const realFilepath = getOnDiskFilepath(options.filePath);
-    const fileDir = dirname(realFilepath);
+    const projectForFile = gqlConfig.getProjectForFile(realFilepath);
+    const schemaKey = projectForFile.schema.toString();
 
-    if (schemaCache.has(fileDir)) {
-      schema = schemaCache.get(fileDir);
+    if (schemaCache.has(schemaKey)) {
+      schema = schemaCache.get(schemaKey);
     } else {
-      const projectForFile = gqlConfig.getProjectForFile(realFilepath);
       schema = projectForFile.getSchemaSync();
-      schemaCache.set(fileDir, schema);
+      schemaCache.set(schemaKey, schema);
     }
   }
 
@@ -50,7 +50,7 @@ export function getSchema(options: ParserOptions, gqlConfig: GraphQLConfig): Gra
   // In this case, the cache key is the path for the schema. This is needed in order to allow separate
   // configurations for different file paths (a very edgey case).
   if (!schema && options.schema) {
-    const schemaKey = Array.isArray(options.schema) ? options.schema.join(',') : options.schema;
+    const schemaKey = options.schema.toString();
 
     if (schemaCache.has(schemaKey)) {
       schema = schemaCache.get(schemaKey);

--- a/packages/plugin/src/schema.ts
+++ b/packages/plugin/src/schema.ts
@@ -8,6 +8,7 @@ import { buildSchema, GraphQLSchema } from 'graphql';
 import { GraphQLConfig } from 'graphql-config';
 import { dirname } from 'path';
 import { ParserOptions } from './types';
+import { getOnDiskFilepath } from './utils';
 
 const schemaCache: Map<string, GraphQLSchema> = new Map();
 
@@ -32,25 +33,23 @@ export function getSchema(options: ParserOptions, gqlConfig: GraphQLConfig): Gra
 
   // We first try to use graphql-config for loading the schema, based on the type of the file,
   // We are using the directory of the file as the key for the schema caching, to avoid reloading of the schema.
-  if (gqlConfig && options?.filePath) {
-    const fileDir = dirname(options.filePath);
+  if (gqlConfig && options.filePath) {
+    const realFilepath = getOnDiskFilepath(options.filePath);
+    const fileDir = dirname(realFilepath);
 
     if (schemaCache.has(fileDir)) {
       schema = schemaCache.get(fileDir);
     } else {
-      const projectForFile = gqlConfig.getProjectForFile(options.filePath);
-
-      if (projectForFile) {
-        schema = projectForFile.getSchemaSync();
-        schemaCache.set(fileDir, schema);
-      }
+      const projectForFile = gqlConfig.getProjectForFile(realFilepath);
+      schema = projectForFile.getSchemaSync();
+      schemaCache.set(fileDir, schema);
     }
   }
 
   // If schema was not loaded yet, and user configured it in the parserConfig, we can try to load it,
   // In this case, the cache key is the path for the schema. This is needed in order to allow separate
   // configurations for different file paths (a very edgey case).
-  if (!schema && options?.schema) {
+  if (!schema && options.schema) {
     const schemaKey = Array.isArray(options.schema) ? options.schema.join(',') : options.schema;
 
     if (schemaCache.has(schemaKey)) {

